cmd/libs/go2idl/go-to-protobuf/protobuf/generator.go

```diff
@@ -117,7 +123,15 @@ func isProtoable(seen map[*types.Type]bool, t *types.Type) bool {
 }
 
 func (g *genProtoIDL) Imports(c *generator.Context) (imports []string) {
-	return g.imports.ImportLines()
+	lines := []string{}
+	// TODO: this could be expressed more cleanly
+	for _, line := range g.imports.ImportLines() {
+		if g.omitGogo && line == "github.com/gogo/protobuf/gogoproto/gogo.proto" {
```

Comment:
This is kind of a hack until we make a decision on supporting protobuf v3 -
the TODO probably covers it for now.

On Fri, Jan 22, 2016 at 2:35 PM, Paul Morie <notifications@github.com>
wrote:

> In cmd/libs/go2idl/go-to-protobuf/protobuf/generator.go
> <https://github.com/kubernetes/kubernetes/pull/19426#discussion_r50580320>
> :
>
> > @@ -117,7 +123,15 @@ func isProtoable(seen map[*types.Type]bool, t *types.Type) bool {
> >  }
> >
> >  func (g *genProtoIDL) Imports(c *generator.Context) (imports []string) {
> > -	return g.imports.ImportLines()
> > +	lines := []string{}
> > +	// TODO: this could be expressed more cleanly
> > +	for _, line := range g.imports.ImportLines() {
> > +		if g.omitGogo && line == "github.com/gogo/protobuf/gogoproto/gogo.proto" {
>
> any possibility of subpackages that need filtering?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19426/files#r50580320>.
>